litellm/proxy/proxy_server.py

```diff
@@ -3106,7 +3115,25 @@ async def completion(
             )
 
         fastapi_response.headers["x-litellm-model-id"] = model_id
-        return response
+        # Try and generate a text completion response in order to comply with tools expecting text not chat responses like langchain text clients (return whatever we got in response if we can't modify it)
+        # todo: refactor as a function in the text completion response class?
+        # todo: streaming response should follow as well
+        try:
+            text_response = TextCompletionResponse()
+            text_response["id"] = response.get("id", None)
+            text_response["object"] = "text_completion"
+            text_response["created"] = response.get("created", None)
+            text_response["model"] = response.get("model", None)
+            text_choices = TextChoices()
+            text_choices["text"] = response["choices"][0]["message"]["content"]
+            text_choices["index"] = response["choices"][0]["index"]
+            text_choices["finish_reason"] = response["choices"][0]["finish_reason"]
+            text_response["choices"] = [text_choices]
+            text_response["usage"] = response.get("usage", None)
+            return text_response
+        except Exception as e:
+            verbose_proxy_logger.debug("Error occurred converting to text completion object - Error:  %s",e)
+            return response
```

Comment:
this should already be getting handled - https://github.com/BerriAI/litellm/blob/22d56037788ca1ad33055768ad99709925c023c4/litellm/main.py#L3168